### PR TITLE
Tickets/dm 5503

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: data/utils/assembleLsstChannels.py [hunk 1]
```diff
@@ -36,16 +36,6 @@
 OutFileName = "image.fits"
```
Comment (PaulPrice) [inline data/utils/assembleLsstChannels.py]: Typo in commit message: "metadat"

Comment (PaulPrice) [inline data/utils/assembleLsstChannels.py]: And I think the commit message needs more explanation.  What are you trying to achieve?

File: python/lsst/obs/test/__init__.py [hunk 2]
```diff
@@ -22,3 +22,4 @@
 from .version import *
 from .testConfig import *
 from .testMapper import *
+from .makeTestRawVisitInfo import *
```
Comment (PaulPrice) [inline python/lsst/obs/test/__init__.py]: Need EOL before EOF.

File: python/lsst/obs/test/makeTestRawVisitInfo.py [hunk 3]
```diff
@@ -64,10 +58,19 @@ def __call__(self, md):
...
         return makeVisitInfo(**argDict)
+
+    def getDateAvg(self, md, exposureTime):
+        """Return date at the middle of the exposure
+
+        @param[in,out] md  FITS metadata; changed in place
+        @param[in] exposureTime  exposure time in sec
+        """
+        startDate = self.popMjdDate(md, "TAI", timesys="TAI")
+        return self.offsetDate(startDate, 0.5*exposureTime)
```
Comment (PaulPrice) [inline python/lsst/obs/test/makeTestRawVisitInfo.py]: I think changes to this file belong with the previous commit.

File: data/utils/assembleLsstRaw.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python2
+#
+# LSST Data Management System
+# Copyright 2014 LSST Corporation.
```
Comment (PaulPrice) [inline data/utils/assembleLsstRaw.py]: Fix year.

File: data/utils/assembleLsstRaw.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    """
+    # views and assembly operations require a masked image, not a DecoratedImage
+    inDecoImage = openChannelImage(dirPath, 0, 0)
+    fullInDim = inDecoImage.getDimensions()
+    yStart = fullInDim[1] - SizeY
+    if yStart < 0:
+        raise RuntimeError("channel image unexpectedly small")
+    subDim = afwGeom.Extent2I(fullInDim[0], SizeY)  # dimensions of the portion of a channel that we use
+    inSubBBox = afwGeom.Box2I(afwGeom.Point2I(0, yStart), subDim)
+    outBBox = afwGeom.Box2I(afwGeom.Point2I(0, 0), subDim * 2)
```
Comment (PaulPrice) [inline data/utils/assembleLsstRaw.py]: No spaces around `*`.

File: data/utils/assembleLsstRaw.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+    print("loading %s as raw unsigned integer data" % (inDecoImageFileName,))
+    return afwImage.DecoratedImageU(inDecoImagePath)
+
+
+def assembleImage(dirPath):
+    """Make one image by combining half of amplifiers C00, C01, C10, C11 of lsstSim data
+    """
+    # views and assembly operations require a masked image, not a DecoratedImage
+    inDecoImage = openChannelImage(dirPath, 0, 0)
```
Comment (PaulPrice) [inline data/utils/assembleLsstRaw.py]: `inDecoImage` is too close to Hungarian notation for my liking, but that's my preference.

File: data/utils/assembleLsstRaw.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            inView = inImage.Factory(inImage, inSubBBox)
+
+            # flip image about x axis for the y = 1 channels
+            if y == 1:
+                arr = inView.getArray()
+                if numpy.any(arr != 0):
+                    arr[:, :] = numpy.flipud(arr)
+
+            xStart = x * subDim[0]
+            yStart = y * subDim[1]
```
Comment (PaulPrice) [inline data/utils/assembleLsstRaw.py]: No space around `*`.

**Comment by r-owen** [comment]
I made all changes except renaming `inDecoImage`. I also improved some commit messages and reordered some commits to increase clarity.

**Pull request merged**